js/spreadsheet.js

```diff
@@ -68,244 +68,196 @@ var spreadsheet = function(options) {
-    var top = row ? row - 1 : row;
-    var bottom = row + 1;
-    // Same as in top variable
-    var left = col ? col - 1 : col;
-    var right = col + 1;
-    // At this block we reacive the nearest cells value from the selected cell
-    var leftValue = hot.getDataAtCell(row, left);
-    var rightValue = hot.getDataAtCell(row, right);
-    var topValue = hot.getDataAtCell(top, col);
-    var bottomValue = hot.getDataAtCell(bottom, col);
-    var dataAt = {
-      left: false,
-      right: false,
-      top: false,
-      bottom: false,
-      all: false,
-      hasData: false
+  function nearestCellsWithData(selectedCell, tableData) {
+    var cellsWithData = {
+      right: tableData[selectedCell.row][selectedCell.col + 1] !== null,
+      left: tableData[selectedCell.row][selectedCell.col - 1] !== null,
+      top: tableData[selectedCell.row - 1][selectedCell.col] !== null,
+      bottom: tableData[selectedCell.row + 1][selectedCell.col] !== null,
     };
+    var countCellsWithData = 0;
 
-    // If there is a data in the selected cell we should select data releated to this cell
-    if (selectedCellData !== null) {
-      dataAt.hasData = true;
-      return dataAt;
+    for (var key in cellsWithData) {
+      if (cellsWithData[key]) {
+        countCellsWithData++;
+      }
     }
-
-    // If no value near the selected cell we should select all table, also fires when we just load data source
-    // and clicked ctrl+a combination
-    if (leftValue === null && rightValue === null && topValue === null && bottomValue === null) {
-      dataAt.all = true;
-      return dataAt;
+    
+    // If there is no true value in cellsWithData object we should select all table
+    if (countCellsWithData === 0) {
+      return {selectAll: true};
     }
 
-    // Showing where is data position from the selected cell
-    dataAt.left = leftValue !== null;
-    dataAt.right = rightValue !== null;
-    dataAt.top = topValue !== null;
-    dataAt.bottom = bottomValue !== null;
-
-    return dataAt;
+    if (tableData[selectedCell.row][selectedCell.col] !== null) {
+      return {cellWithData: true};
+    }
+    
+    return cellsWithData;
   }
 
   /**
-   * Method to get a coordinats which we need to select
-   * 
-   * @param {Array} startAt - array of the selected coordinats 
-   * @param {Object} moveTo - object that returned from closestData() function
+   * The function that helps us to find start index for further selection
    * 
-   * @returns {Array} - coordinats that needs to be selected. Example of the returned data: [[startRow, startCol, endRow, endCol]]
+   * @param {array} tableData - an array of data from the Handsontable library
+   * @param {int} startSearchFrom - index of the row/col from what we should start a search
+   * @param {int} colIndex - a not necessary option that tells us what we need to look, a row index or column index
```

Comment:
Am I right in suggesting this?

```suggestion
   * @param {int} colIndex - (optional) when provided, the search is performed along the rows for the column specified
```